server/src/main/java/org/elasticsearch/action/get/TransportGetAction.java

```diff
@@ -79,19 +75,7 @@ protected ShardIterator shards(ClusterState state, InternalRequest request) {
                 request.request().routing(),
                 request.request().preference()
             );
-        if (iterator == null) {
-            return null;
-        }
-        // If it is stateless, only route isPromotableToPrimary shards. This is a temporary workaround until a more cohesive solution can be
-        // implemented for search shards.
-        if (DiscoveryNode.isStateless(clusterService.getSettings())) {
-            return new PlainShardIterator(
-                iterator.shardId(),
-                iterator.getShardRoutings().stream().filter(ShardRouting::isPromotableToPrimary).collect(Collectors.toList())
-            );
-        } else {
-            return iterator;
-        }
+        return clusterService.operationRouting().useOnlyPromotableShardsForStateless(iterator);
```

Comment:
@dnhatn are these changes to Get and MGet necessary for correctly routing term vector requests? Or is it just to make use of `useOnlyPromotableShardsForStateless`? I need to change this to route gets to search shards, and I am wondering whether this is needed/shared with term vectors or not.

Reply:
@pxsalehi The term vectors API just uses `useOnlyPromotableShardsForStateless` shards. Feel free to make changes to the get and mget APIs.